tests/drivers/i2c/i2c_emul/src/emulated_target.cpp

```diff
@@ -42,7 +42,7 @@ struct i2c_target_callbacks emulated_callbacks[FORWARD_COUNT] = {
 	DT_FOREACH_PROP_ELEM(CONTROLLER_LABEL, forwards, DEFINE_EMULATED_CALLBACK)};
 
 #define DEFINE_EMULATED_TARGET_CONFIG(node_id, prop, n)                                            \
-	[n] = {                                                                                    \
+	{                                                                                          \
```

Comment:
This looks like its doing something completely different than it was before? Confused by this change, why is "[n] = " no longer needed, why was it needed before?

Reply:
no idea why it was introduced originally, I guess to do something with that `n` variable? regardless, the compiler does not like it, the alternative would be to shut off that warning, happy with either. @yperess?